### PR TITLE
new queue, ack format, misc js

### PR DESCRIPTION
**Pull request by scottf**
Catching up to server v2.4.0 Major queueing overhaul, preparing for new ack format (taken from another pr), misc cleanup.

### CODE REVIEW

File: src/NATS.Client/Internals/JsonUtils.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ internal static List<string> OptionalStringList(JSONNode node, String field)
         }
         
         internal static byte[] AsByteArrayFromBase64(JSONNode node) {
-            return string.IsNullOrEmpty(node.Value) ? null : Convert.FromBase64String(node.Value);
+            return Validator.NullOrEmpty(node.Value) ? null : Convert.FromBase64String(node.Value);
```
Comment (scottf) [inline src/NATS.Client/Internals/JsonUtils.cs]: "   " is not empty for string.IsNullOrEmpty so use Validator version

Comment (sixlettervariables) [reply src/NATS.Client/Internals/JsonUtils.cs]: This should be `String.IsNullOrWhiteSpace`.

File: src/NATS.Client/Internals/SimpleJson.cs [hunk 2]
```diff
@@ -1036,7 +1036,7 @@ internal partial class JSONString : JSONNode
 
         public override JSONNodeType Tag { get { return JSONNodeType.String; } }
         public override bool IsString { get { return true; } }
-        public override bool ShouldWrite { get { return !string.IsNullOrEmpty(m_Data); } }
+        public override bool ShouldWrite { get { return !Validator.NullOrEmpty(m_Data); } }
```
Comment (sixlettervariables) [inline src/NATS.Client/Internals/SimpleJson.cs]: `Steing.IsNullOrWhiteSpace`

File: src/NATS.Client/JetStream/JetStream.cs [hunk 3]
```diff
@@ -176,27 +182,47 @@ public Task<PublishAck> PublishAsync(Msg msg, PublishOptions publishOptions)
...
+                            }
+                        }
+                        else { // else they requested a queue but this durable was not configured as queue
+                            throw new ArgumentException($"Existing consumer [{durable}] is not configured as a queue / deliver group.");
+                        }
+                    }
+                    else if (queueName == null) {
+                        throw new ArgumentException($"Existing consumer [{durable}] is configured as a queue / deliver group.");
+                    }
+                    else if (!lookedUp.Equals(queueName)) {
```
Comment (sixlettervariables) [inline src/NATS.Client/JetStream/JetStream.cs]: Using `Equals` with strings without passing a `StringComparison` is odd. Just use `==`

**Review by sixlettervariables — DISMISSED** [review]

File: src/NATS.Client/JetStream/JetStream.cs (hunk 3 above)
Comment (scottf) [reply src/NATS.Client/JetStream/JetStream.cs]: okay that's a java habit, will fix

File: src/NATS.Client/Internals/SimpleJson.cs (hunk 2 above)
Comment (scottf) [reply src/NATS.Client/Internals/SimpleJson.cs]: gotcha

File: src/NATS.Client/Internals/JsonUtils.cs (hunk 1 above)
Comment (scottf) [reply src/NATS.Client/Internals/JsonUtils.cs]: switching to IsNullOrWhiteSpace

**Pull request merged**